src/main/java/de/wi2020sebgroup1/cinema/controller/CinemaController.java

```diff
@@ -0,0 +1,82 @@
+package de.wi2020sebgroup1.cinema.controller;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.wi2020sebgroup1.cinema.entities.Cinema;
+import de.wi2020sebgroup1.cinema.entities.City;
+import de.wi2020sebgroup1.cinema.entities.User;
+import de.wi2020sebgroup1.cinema.repositories.CinemaRepository;
+import de.wi2020sebgroup1.cinema.repositories.CityRepository;
+import de.wi2020sebgroup1.cinema.repositories.UserRepository;
+
+@Controller
+@RestController
+@RequestMapping("/cinema")
+public class CinemaController {
+	
+	@Autowired
+	CinemaRepository cinemaRepository;
+	
+	@Autowired
+	CityRepository cityRepository;
+	
+	@PutMapping("/add")
+	public ResponseEntity<Cinema> addCinema(@RequestBody Cinema newCinema){
+		
+		Cinema toAddCinema = newCinema;
+		City toCheckCity = newCinema.getCity();
+		
+		List<City> toGetCity = cityRepository.findByPlz(toCheckCity.getPlz());
+		if(toGetCity.size() == 1) {
+			toAddCinema.setCity(toGetCity.get(0));
+		}
+		
+		return new ResponseEntity<>( cinemaRepository.save(toAddCinema), HttpStatus.CREATED);
+	}
+	
+	@GetMapping("/getAll")
+	public ResponseEntity<Iterable<Cinema>> getCinemas(){
+		return new ResponseEntity<>(cinemaRepository.findAll(),HttpStatus.FOUND);
+	}
+	
+	@GetMapping("/{id}")
+	public ResponseEntity<Cinema> getCinema(@PathVariable UUID id){
+		Optional<Cinema> search = cinemaRepository.findById(id);
+		try {
+			Cinema found = search.get();
+			return new ResponseEntity<>(found, HttpStatus.FOUND);
+		}catch(NoSuchElementException e) {
+			return new ResponseEntity<>(null, HttpStatus.NOT_FOUND);
+		}
+		
+	}
+	
+	@DeleteMapping("/{id}")
```

Comment:
Wollen wir lieber so URLs nehmen wie /admin/delete/ID? Weil dann kann man diese URLs nur für admins freigeben lassen in der Security Config

Reply:
Hmmm dann bräuchten wir einen Admin-Controller, aber diese Funktionen sind sowieso nur durch ein Login ausführbar und auch ein Create sollte ohne Admin nicht möglich sein (Admin wie auch die Controller der Website). Bin mir also unsicher, wie würdest du das angehen?

Reply:
In dem Beispiel von Jonas ist das so umgesetzt, die haben eine etwas kompliziertere User Controller/Service Architektur, die ich noch versuche zu verstehen. Und da kann man dann halt /admin/[...] eingeben und nur eingeloggte Admins machen, weil es dafür ne Pfadgruppe gibt, die dann über die Authentifizierung zugelassen/gesperrt wird.

Reply:
Ist auch in vielen Tutorials so umgesetzt

Reply:
Können wir Problemlos über einen Admin Controller machen. 
Dann müssten wir die ganzen Aktionen nur in diesem Admin Controller unterbringen.

tbh die Delete funktioniert sowieso noch nicht

Reply:
Nice, dann machen wir das so :)